Source/JavaScriptCore/bytecode/AccessCase.cpp

```diff
@@ -1210,32 +1357,56 @@ void AccessCase::generateWithGuard(
     case IndexedTypedArrayInt32Load:
     case IndexedTypedArrayUint32Load:
     case IndexedTypedArrayFloat32Load:
-    case IndexedTypedArrayFloat64Load: {
+    case IndexedTypedArrayFloat64Load:
+    case IndexedResizableTypedArrayInt8Load:
+    case IndexedResizableTypedArrayUint8Load:
+    case IndexedResizableTypedArrayUint8ClampedLoad:
+    case IndexedResizableTypedArrayInt16Load:
+    case IndexedResizableTypedArrayUint16Load:
+    case IndexedResizableTypedArrayInt32Load:
+    case IndexedResizableTypedArrayUint32Load:
+    case IndexedResizableTypedArrayFloat32Load:
+    case IndexedResizableTypedArrayFloat64Load: {
         ASSERT(!viaProxy());
         // This code is written such that the result could alias with the base or the property.
 
         TypedArrayType type = toTypedArrayType(m_type);
+        bool isResizableOrGrowableSharedMode = forResizableTypedArray(m_type);
 
         GPRReg propertyGPR = stubInfo.propertyGPR();
 
         fallThrough.append(jit.branch8(CCallHelpers::NotEqual, CCallHelpers::Address(baseGPR, JSCell::typeInfoTypeOffset()), CCallHelpers::TrustedImm32(typeForTypedArrayType(type))));
-        fallThrough.append(jit.branchTest8(CCallHelpers::NonZero, CCallHelpers::Address(baseGPR, JSArrayBufferView::offsetOfMode()), CCallHelpers::TrustedImm32(isResizableOrGrowableSharedMode)));
 
-        CCallHelpers::Address addressOfLength = CCallHelpers::Address(baseGPR, JSArrayBufferView::offsetOfLength());
-        jit.signExtend32ToPtr(propertyGPR, scratchGPR);
+        if (!isResizableOrGrowableSharedMode) {
+            fallThrough.append(jit.branchTest8(CCallHelpers::NonZero, CCallHelpers::Address(baseGPR, JSArrayBufferView::offsetOfMode()), CCallHelpers::TrustedImm32(isResizableOrGrowableSharedMode)));
```

Comment:
I guess this could just be a constant now?
```suggestion
            fallThrough.append(jit.branchTest8(CCallHelpers::NonZero, CCallHelpers::Address(baseGPR, JSArrayBufferView::offsetOfMode()), CCallHelpers::TrustedImm32(false)));
```

Reply:
Oops, this is a bug. `isResizableOrGrowableSharedMode` should be renamed to isResizableOrGrowableShared.